src/overlays/actors/ovl_En_Ani/z_en_ani.c

```diff
@@ -21,39 +45,311 @@ const ActorInit En_Ani_InitVars = {
+        switch (this->unk_2AA) {
+            case 0:
+                func_809B0A6C(this, globalCtx);
+                break;
+            case 1:
+                func_809B0A28(this, globalCtx);
+                break;
+            case 2:
+                func_809B0994(this, globalCtx);
+                break;
+            case 3:
+                func_809B0A28(this, globalCtx);
+                break;
+            case 4:
+                func_809B0988(this, globalCtx);
+                break;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ani/func_809B0558.s")
+        if (globalCtx->csCtx.frames == 0x64) {
+            func_800788CC(NA_SE_IT_EARTHQUAKE);
+        }
+    } else {
+        if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+            this->skelAnime.animCurrentFrame = 0.0f;
+        }
+        this->actionFunc(this, globalCtx);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ani/func_809B05F0.s")
+    if ((this->unk_2A8 & 1) != 0) {
+        func_80038290(globalCtx, &this->actor, &this->unk_29C, &this->unk_2A2, this->actor.posRot2.pos);
+        this->unk_2A2.z = 0;
+        this->unk_2A2.y = this->unk_2A2.z;
+        this->unk_2A2.x = this->unk_2A2.z;
+    } else {
+        Math_SmoothScaleMaxMinS(&this->unk_29C.x, 0, 6, 6200, 100);
+        Math_SmoothScaleMaxMinS(&this->unk_29C.y, 0, 6, 6200, 100);
+        Math_SmoothScaleMaxMinS(&this->unk_2A2.x, 0, 6, 6200, 100);
+        Math_SmoothScaleMaxMinS(&this->unk_2A2.y, 0, 6, 6200, 100);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ani/func_809B064C.s")
+    if (this->unk_2AE == 0) {
+        phi_v0 = 0;
+    } else {
+        this->unk_2AE -= 1;
+        phi_v0 = this->unk_2AE;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ani/func_809B07F8.s")
+    if (phi_v0 == 0) {
+        this->unk_2AE = Math_Rand_S16Offset(0x3c, 0x3c);
+    }
+    this->unk_2AC = this->unk_2AE;
+    if (this->unk_2AC >= 3) {
+        this->unk_2AC = 0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ani/func_809B0988.s")
+s32 EnAni_UpdateMtxFunc(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, EnAni* enAni) {
+    EnAni* temp;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ani/func_809B0994.s")
+    temp = enAni;
+    if (limbIndex == 0xf) {
+        rot->x += temp->unk_29C.y;
+        rot->z += temp->unk_29C.x;
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ani/func_809B0A28.s")
+void EnAni_LimbAppendDList(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, EnAni* enAni) {
+    if (limbIndex == 0xf) {
+        Matrix_MultVec3f(&EnAniVec, &enAni->actor.posRot2.pos);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ani/func_809B0A6C.s")
+void EnAni_Draw(EnAni* this, GlobalContext* globalCtx) {
+    UNK_PTR* temp;
```

Comment:
I think i do need the star here, because temp is being set to an array of type UNK_PTR (defined on line 72). it doesnt compile if i get rid of the star

Reply:
oh I see, nevermind then